src/eval.rs

```diff
@@ -380,8 +379,8 @@ impl Scope {
     }
 
     pub fn replace(&mut self, name: BString, val: ValRef) -> bool {
-        if self.map.contains_key(&name) {
-            self.map.insert(name, val);
+        if let Entry::Occupied(mut e) = self.map.entry(name.clone()) {
+            e.insert(val);
```

Comment:
Is it possible to do this without cloning the `name`?

Reply:
No, sadly not, I tried it, but it didn't work.